Spiel startet nicht

Beim Starten eines Spiels zwischen dem Beispiel-Computerspielers und einem Menschen passiert nichts, außer dass "Das Spiel startet... Bitte verbinde manuell gestartete Clients auf localhost:13050" angezeigt wird.

Java:
```
openjdk version "16.0.2" 2021-07-20
OpenJDK Runtime Environment (build 16.0.2+7)
OpenJDK 64-Bit Server VM (build 16.0.2+7, mixed mode)
```

System:
```
5.14.3-arch1-1 x86_64 GNU/Linux
```

Logs:
```
21:17:04,535 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [/home/logback.xml]
21:17:04,536 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [jar:file:/home/.cache/socha/socha.jar!/logback-test.xml]
21:17:04,548 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@196debbd - URL [jar:file:/home/.cache/socha/socha.jar!/logback-test.xml] is not of type file
21:17:04,679 |-INFO in ch.qos.logback.core.joran.action.TimestampAction - Using current interpretation time, i.e. now, as time reference.
21:17:04,680 |-INFO in ch.qos.logback.core.joran.action.TimestampAction - Adding property to the context with key="time" and value="09-17T211704" to the LOCAL scope
21:17:04,680 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [sc.util.TidyFileAppender]
21:17:04,687 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - This appender no longer admits a layout as a sub-component, set an encoder instead.
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details
21:17:04,718 |-INFO in sc.util.TidyFileAppender[FILE] - File property is set to [log/game-server_09-17T211704.log]
21:17:04,719 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
21:17:04,720 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
21:17:04,721 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.PatternLayout] for [layout] property
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - This appender no longer admits a layout as a sub-component, set an encoder instead.
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc] to DEBUG
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc.networking.clients.XStreamClient] to INFO
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc.gui] to TRACE
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
21:17:04,723 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
21:17:04,723 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
21:17:04,724 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
21:17:04,724 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@50cbde79 - Registering current configuration as safe fallback point
Sept. 17, 2021 9:17:04 PM tornadofx.Stylesheet$Companion detectAndInstallUrlHandler
INFORMATION: Installing CSS url handler, since it was not picked up automatically
21:17:04,535 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [/home/logback.xml]
21:17:04,536 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback-test.xml] at [jar:file:/home/.cache/socha/socha.jar!/logback-test.xml]
21:17:04,548 |-INFO in ch.qos.logback.core.joran.spi.ConfigurationWatchList@196debbd - URL [jar:file:/home/.cache/socha/socha.jar!/logback-test.xml] is not of type file
21:17:04,679 |-INFO in ch.qos.logback.core.joran.action.TimestampAction - Using current interpretation time, i.e. now, as time reference.
21:17:04,680 |-INFO in ch.qos.logback.core.joran.action.TimestampAction - Adding property to the context with key="time" and value="09-17T211704" to the LOCAL scope
21:17:04,680 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [sc.util.TidyFileAppender]
21:17:04,687 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - This appender no longer admits a layout as a sub-component, set an encoder instead.
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
21:17:04,718 |-WARN in sc.util.TidyFileAppender[FILE] - See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details
21:17:04,718 |-INFO in sc.util.TidyFileAppender[FILE] - File property is set to [log/game-server_09-17T211704.log]
21:17:04,719 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
21:17:04,720 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [STDOUT]
21:17:04,721 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.PatternLayout] for [layout] property
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - This appender no longer admits a layout as a sub-component, set an encoder instead.
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
21:17:04,722 |-WARN in ch.qos.logback.core.ConsoleAppender[STDOUT] - See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc] to DEBUG
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc.networking.clients.XStreamClient] to INFO
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [sc.gui] to TRACE
21:17:04,723 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
21:17:04,723 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
21:17:04,723 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [STDOUT] to Logger[ROOT]
21:17:04,724 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
21:17:04,724 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@50cbde79 - Registering current configuration as safe fallback point

21:17:04 WARN               sc.server.Configuration - Could not find server.properties at /home/server.properties, will use default values!
21:17:04 INFO   sc.server.network.NewClientListener - Listening on port 13050 for incoming connections.
21:17:04 DEBUG             sc.server.ServiceManager - Spawning thread for new service (name=NewClientListener, daemon=true)
21:17:04 DEBUG             sc.server.ServiceManager - Spawning thread for new service (name=ClientManager, daemon=true)
21:17:04 INFO       sc.server.network.ClientManager - ClientManager running
21:17:07 DEBUG      sc.gui.controller.AppController - Requested View change from START -> GAME_CREATION
21:17:11 DEBUG      sc.gui.controller.AppController - Requested View change from GAME_CREATION -> GAME_LOADING
21:17:11 INFO   sc.networking.clients.XStreamClient - Creating TCP Network for localhost:13050
21:17:11 INFO   sc.server.network.NewClientListener - A Client connected...
21:17:11 INFO   sc.server.network.NewClientListener - Added Client Client@47124b1a to ReadyQueue.
21:17:11 INFO       sc.server.network.ClientManager - Delegating new client to ClientManager...
21:17:11 INFO       sc.server.network.ClientManager - Delegation done
21:17:11 INFO   sc.networking.clients.XStreamClient - Creating TCP Network for localhost:13050
21:17:11 INFO   sc.server.network.NewClientListener - A Client connected...
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
21:17:11 INFO              sc.server.network.Client - Client authenticated as administrator
21:17:11 INFO   sc.networking.clients.XStreamClient - Creating TCP Network for localhost:13050
21:17:11 INFO   sc.server.network.NewClientListener - Added Client Client@251468fa to ReadyQueue.
21:17:11 INFO       sc.server.network.ClientManager - Delegating new client to ClientManager...
21:17:11 INFO       sc.server.network.ClientManager - Delegation done
21:17:11 INFO   sc.server.network.NewClientListener - A Client connected...
21:17:11 DEBUG                  sc.gui.LobbyManager - Starting new game (paused: false, players: [Player(name=Spieler 1, client=sc.gui.controller.client.GuiClient@2390520f type Beispiel-Computerspieler on localhost:13050), Player(name=Spieler 2, client=sc.gui.controller.client.GuiClient@28b15fde type Mensch on localhost:13050)])
21:17:11 INFO       sc.server.network.ClientManager - Delegating new client to ClientManager...
21:17:11 INFO       sc.server.network.ClientManager - Delegation done
21:17:11 INFO   sc.server.network.NewClientListener - Added Client Client@420a6ac2 to ReadyQueue.
21:17:11 DEBUG     sc.server.gaming.GameRoomManager - Adding room with id 346cf859-52c6-48d3-911a-0c0096bebcf0
21:17:11 WARN             sc.server.gaming.GameRoom - PAUSE is already false, dropping request
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
21:17:11 INFO   sc.server.gaming.ReservationManager - Reservation 362ee7bd-d4dc-422f-9a04-91721ccae823 was redeemed.
21:17:11 DEBUG            sc.server.gaming.GameRoom - startIfReady called
21:17:11 INFO             sc.server.gaming.GameRoom - Game not ready yet: Game(players=[ONE(Spieler 1), TWO(Spieler 2)], gameState=GameState R0T0 -> Rot (Bernsteine {}))
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
21:17:11 INFO   sc.server.gaming.ReservationManager - Reservation a4e5cf6f-51d0-4360-a866-b208bb0dbe5c was redeemed.
21:17:11 DEBUG            sc.server.gaming.GameRoom - startIfReady called
21:17:11 DEBUG    sc.framework.plugins.AbstractGame - Notifying sc.server.gaming.GameRoom@5c9bda10 about new game state
Security framework of XStream not explicitly initialized, using predefined black list on your own risk.
21:17:11 ERROR    sc.framework.plugins.AbstractGame - NewState Notification caused an exception
java.lang.NullPointerException: Cannot invoke "Object.toString()" because the return value of "java.util.Map$Entry.getValue()" is null
	at com.thoughtworks.xstream.core.DefaultConverterLookup.lookupConverterForType(DefaultConverterLookup.java:96)
	at com.thoughtworks.xstream.XStream$1.lookupConverterForType(XStream.java:485)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:48)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshallField(AbstractReflectionConverter.java:270)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.writeField(AbstractReflectionConverter.java:174)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doMarshal(AbstractReflectionConverter.java:262)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:90)
	at com.thoughtworks.xstream.core.TreeMarshaller.convert(TreeMarshaller.java:70)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshallField(AbstractReflectionConverter.java:270)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.writeField(AbstractReflectionConverter.java:174)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doMarshal(AbstractReflectionConverter.java:262)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:90)
	at com.thoughtworks.xstream.core.TreeMarshaller.convert(TreeMarshaller.java:70)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshallField(AbstractReflectionConverter.java:270)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter$2.writeField(AbstractReflectionConverter.java:174)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doMarshal(AbstractReflectionConverter.java:262)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.marshal(AbstractReflectionConverter.java:90)
	at com.thoughtworks.xstream.core.TreeMarshaller.convert(TreeMarshaller.java:70)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:58)
	at com.thoughtworks.xstream.core.TreeMarshaller.convertAnother(TreeMarshaller.java:43)
	at com.thoughtworks.xstream.core.TreeMarshaller.start(TreeMarshaller.java:82)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.marshal(AbstractTreeMarshallingStrategy.java:37)
	at com.thoughtworks.xstream.XStream.marshal(XStream.java:1278)
	at com.thoughtworks.xstream.XStream$3.writeToStream(XStream.java:2023)
	at com.thoughtworks.xstream.core.util.CustomObjectOutputStream.writeObjectOverride(CustomObjectOutputStream.java:87)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
	at sc.networking.clients.XStreamClient.sendObject(XStreamClient.java:187)
	at sc.networking.clients.XStreamClient.send(XStreamClient.java:168)
	at sc.server.gaming.GameRoom.lambda$observerBroadcast$3(GameRoom.java:132)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at sc.server.gaming.GameRoom.observerBroadcast(GameRoom.java:132)
	at sc.server.gaming.GameRoom.onStateChanged(GameRoom.java:145)
	at sc.framework.plugins.AbstractGame.notifyOnNewState(AbstractGame.kt:187)
	at sc.framework.plugins.AbstractGame.next(AbstractGame.kt:108)
	at sc.framework.plugins.AbstractGame.start(AbstractGame.kt:98)
	at sc.server.gaming.GameRoom.start(GameRoom.java:229)
	at sc.server.gaming.GameRoom.startIfReady(GameRoom.java:225)
	at sc.server.gaming.GameRoom.fillSlot(GameRoom.java:203)
	at sc.server.gaming.ReservationManager.redeemReservationCode(ReservationManager.java:38)
	at sc.server.Lobby.onRequest(Lobby.kt:45)
	at sc.server.network.Client.onObject(Client.java:110)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:114)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 INFO     sc.framework.plugins.AbstractGame - Sending MoveRequest to player ONE(Spieler 1)
21:17:11 DEBUG          sc.framework.plugins.Player - Move requested from ONE(Spieler 1)
21:17:11 INFO             sc.server.gaming.GameRoom - Updating Status to ACTIVE (was: CREATED)
21:17:11 ERROR  sc.networking.clients.XStreamClient - Unknown Communication Error
java.lang.IllegalStateException: Received move request before GameState!
	at sc.gui.controller.client.InternalGameHandler.calculateMove(GuiClient.kt:37)
	at sc.player.PlayerClient.apply(PlayerClient.kt:31)
	at sc.player.PlayerClient.apply(PlayerClient.kt:24)
	at sc.networking.clients.LobbyClient.onObject(LobbyClient.java:85)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:114)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 WARN   sc.networking.clients.XStreamClient - LobbyClient@3bc1cb03 disconnected (Cause: UNKNOWN, Exception: {})
java.lang.IllegalStateException: Received move request before GameState!
	at sc.gui.controller.client.InternalGameHandler.calculateMove(GuiClient.kt:37)
	at sc.player.PlayerClient.apply(PlayerClient.kt:31)
	at sc.player.PlayerClient.apply(PlayerClient.kt:24)
	at sc.networking.clients.LobbyClient.onObject(LobbyClient.java:85)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:114)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 WARN   sc.networking.clients.XStreamClient - Client@251468fa disconnected (Cause: LOST_CONNECTION, Exception: {})
java.io.EOFException: null
	at com.thoughtworks.xstream.XStream$4.readFromStream(XStream.java:2102)
	at com.thoughtworks.xstream.core.util.CustomObjectInputStream.readObjectOverride(CustomObjectInputStream.java:123)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:486)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:453)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:102)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 INFO   sc.networking.clients.XStreamClient - Stopping XStream-Receive id:28 of LobbyClient@3bc1cb03
21:17:11 INFO   sc.networking.clients.XStreamClient - Stopping XStream-Receive id:27 of Client@251468fa
21:17:11 WARN   sc.networking.clients.XStreamClient - ReceiveThread is stopping itself
21:17:11 WARN   sc.networking.clients.XStreamClient - ReceiveThread is stopping itself
21:17:11 INFO             sc.server.gaming.GameRoom - Started Game(players=[ONE(Spieler 1), TWO(Spieler 2)], gameState=GameState R0T0 -> Rot (Bernsteine {}))
21:17:11 DEBUG             sc.networking.TcpNetwork - Closing TcpNetwork{socket=Socket[addr=localhost/127.0.0.1,port=13050,localport=59652]}
21:17:11 DEBUG             sc.networking.TcpNetwork - Closing TcpNetwork{socket=Socket[addr=/127.0.0.1,port=59652,localport=13050]}
21:17:11 INFO       sc.server.network.ClientManager - Removing client Client@251468fa from client manager
21:17:11 INFO             sc.server.gaming.GameRoom - Removing ONE(Spieler 1) from sc.server.gaming.GameRoom@5c9bda10
21:17:11 INFO     sc.framework.plugins.AbstractGame - Stopping Game(players=[ONE(Spieler 1), TWO(Spieler 2)], gameState=GameState R0T0 -> Rot (Bernsteine {}))
21:17:11 INFO    sc.framework.plugins.ActionTimeout - HardTimout wasn't reached.
21:17:11 DEBUG                   sc.plugin2022.Game - Calculating score for ONE(Spieler 1)
21:17:11 DEBUG                   sc.plugin2022.Game - Calculating score for TWO(Spieler 2)
21:17:11 INFO             sc.server.gaming.GameRoom - Updating Status to OVER (was: ACTIVE)
21:17:11 INFO             sc.server.gaming.GameRoom - Game(players=[ONE(Spieler 1), TWO(Spieler 2)], gameState=GameState R0T0 -> Rot (Bernsteine {})) is over (regular=false)
21:17:11 WARN   sc.networking.clients.XStreamClient - Writing on a closed Stream -> dropped the packet (tried to send package of type RoomPacket)
21:17:11 WARN   sc.networking.clients.XStreamClient - LobbyClient@a6b701c disconnected (Cause: PROTOCOL_ERROR, Exception: {})
com.thoughtworks.xstream.converters.ConversionException: No converter available
---- Debugging information ----
message             : No converter available
type                : java.util.EnumMap
converter           : com.thoughtworks.xstream.converters.enums.EnumMapConverter
message[1]          : Could not initialize class com.thoughtworks.xstream.converters.enums.EnumMapConverter$Reflections
converter[1]        : com.thoughtworks.xstream.converters.reflection.SerializableConverter
message[2]          : Unable to make private void java.util.EnumMap.readObject(java.io.ObjectInputStream) throws java.io.IOException,java.lang.ClassNotFoundException accessible: module java.base does not "opens java.util" to unnamed module @5516c33a
converter[2]        : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
message[3]          : Unable to make field transient java.util.Set java.util.AbstractMap.keySet accessible: module java.base does not "opens java.util" to unnamed module @5516c33a
class               : sc.plugin2022.GameState
required-type       : sc.plugin2022.GameState
converter-type      : com.thoughtworks.xstream.converters.reflection.ReflectionConverter
line number         : 79
class[1]            : sc.protocol.room.MementoMessage
required-type[1]    : sc.protocol.room.MementoMessage
class[2]            : sc.protocol.room.RoomPacket
required-type[2]    : sc.protocol.room.RoomPacket
version             : not available
-------------------------------
	at com.thoughtworks.xstream.core.DefaultConverterLookup.lookupConverterForType(DefaultConverterLookup.java:88)
	at com.thoughtworks.xstream.XStream$1.lookupConverterForType(XStream.java:485)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:56)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshallField(AbstractReflectionConverter.java:499)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.doUnmarshal(AbstractReflectionConverter.java:425)
	at com.thoughtworks.xstream.converters.reflection.AbstractReflectionConverter.unmarshal(AbstractReflectionConverter.java:277)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convert(TreeUnmarshaller.java:72)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:66)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.convertAnother(TreeUnmarshaller.java:50)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.start(TreeUnmarshaller.java:134)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.unmarshal(AbstractTreeMarshallingStrategy.java:32)
	at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1431)
	at com.thoughtworks.xstream.XStream$4.readFromStream(XStream.java:2105)
	at com.thoughtworks.xstream.core.util.CustomObjectInputStream.readObjectOverride(CustomObjectInputStream.java:123)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:486)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:453)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:102)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 INFO   sc.networking.clients.XStreamClient - Stopping XStream-Receive id:26 of LobbyClient@a6b701c
21:17:11 WARN   sc.networking.clients.XStreamClient - ReceiveThread is stopping itself
21:17:11 DEBUG             sc.networking.TcpNetwork - Closing TcpNetwork{socket=Socket[addr=localhost/127.0.0.1,port=13050,localport=59650]}
21:17:11 WARN   sc.networking.clients.XStreamClient - Client@47124b1a disconnected (Cause: LOST_CONNECTION, Exception: {})
java.io.EOFException: null
	at com.thoughtworks.xstream.XStream$4.readFromStream(XStream.java:2102)
	at com.thoughtworks.xstream.core.util.CustomObjectInputStream.readObjectOverride(CustomObjectInputStream.java:123)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:486)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:453)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:102)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
21:17:11 INFO     sc.networking.clients.LobbyClient - Received game result: GameResult(winner=null, scores=[Spieler 1(cause=LEFT, reason='Der Spieler hat das Spiel verlassen', parts=[Siegpunkte=0, Bernsteine=0, Figur vorne=6]), Spieler 2(cause=REGULAR, reason='', parts=[Siegpunkte=2, Bernsteine=0, Figur vorne=7])])
21:17:11 WARN   sc.networking.clients.XStreamClient - Client@47124b1a disconnected (Cause: PROTOCOL_ERROR, Exception: {})
com.thoughtworks.xstream.io.StreamException: 
	at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:77)
	at com.thoughtworks.xstream.io.xml.PrettyPrintWriter.flush(PrettyPrintWriter.java:346)
	at com.thoughtworks.xstream.io.WriterWrapper.flush(WriterWrapper.java:49)
	at com.thoughtworks.xstream.io.StatefulWriter.flush(StatefulWriter.java:139)
	at com.thoughtworks.xstream.XStream$3.flush(XStream.java:2035)
	at com.thoughtworks.xstream.core.util.CustomObjectOutputStream.flush(CustomObjectOutputStream.java:145)
	at sc.networking.clients.XStreamClient.sendObject(XStreamClient.java:188)
	at sc.networking.clients.XStreamClient.send(XStreamClient.java:168)
	at sc.server.gaming.GameRoom.lambda$broadcast$2(GameRoom.java:127)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at sc.server.gaming.GameRoom.broadcast(GameRoom.java:127)
	at sc.server.gaming.GameRoom.broadcast(GameRoom.java:121)
	at sc.server.gaming.GameRoom.onGameOver(GameRoom.java:76)
	at sc.framework.plugins.AbstractGame.notifyOnGameOver(AbstractGame.kt:176)
	at sc.framework.plugins.AbstractGame.stop(AbstractGame.kt:92)
	at sc.server.gaming.GameRoom.cancel(GameRoom.java:382)
	at sc.server.gaming.GameRoom.removePlayer(GameRoom.java:411)
	at sc.server.gaming.PlayerSlot.onClientDisconnected(PlayerSlot.java:90)
	at sc.server.network.Client.onDisconnect(Client.java:89)
	at sc.networking.clients.XStreamClient.handleDisconnect(XStreamClient.java:215)
	at sc.networking.clients.XStreamClient.receiveThread(XStreamClient.java:127)
	at sc.networking.clients.XStreamClient$1.run(XStreamClient.java:79)
	at java.base/java.lang.Thread.run(Thread.java:831)
Caused by: java.net.SocketException: Socket closed
	at java.base/sun.nio.ch.NioSocketImpl.ensureOpenAndConnected(NioSocketImpl.java:165)
	at java.base/sun.nio.ch.NioSocketImpl.beginWrite(NioSocketImpl.java:366)
	at java.base/sun.nio.ch.NioSocketImpl.implWrite(NioSocketImpl.java:411)
	at java.base/sun.nio.ch.NioSocketImpl.write(NioSocketImpl.java:440)
	at java.base/sun.nio.ch.NioSocketImpl$2.write(NioSocketImpl.java:826)
	at java.base/java.net.Socket$SocketOutputStream.write(Socket.java:1045)
	at java.base/sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:242)
	at java.base/sun.nio.cs.StreamEncoder.implFlushBuffer(StreamEncoder.java:321)
	at java.base/sun.nio.cs.StreamEncoder.implFlush(StreamEncoder.java:325)
	at java.base/sun.nio.cs.StreamEncoder.flush(StreamEncoder.java:159)
	at java.base/java.io.OutputStreamWriter.flush(OutputStreamWriter.java:248)
	at com.thoughtworks.xstream.core.util.QuickWriter.flush(QuickWriter.java:75)
	... 22 common frames omitted
21:17:11 INFO   sc.networking.clients.XStreamClient - Stopping XStream-Receive id:25 of Client@47124b1a
21:17:11 DEBUG             sc.networking.TcpNetwork - Closing TcpNetwork{socket=Socket[addr=/127.0.0.1,port=59650,localport=13050]}
21:17:11 INFO       sc.server.network.ClientManager - Removing client Client@47124b1a from client manager
21:17:11 WARN   sc.networking.clients.XStreamClient - Attempted to close an already closed stream
21:17:11 DEBUG            sc.server.gaming.GameRoom - Saving replay to /home/replays/replay_swc_2022_ostseeschach_2021-09-17_21-17-11_Spieler_1_Spieler_2.xml
21:17:11 DEBUG            sc.server.gaming.GameRoom - Kicking clients and observers
21:17:11 WARN   sc.networking.clients.XStreamClient - Writing on a closed Stream -> dropped the packet (tried to send package of type RemovedFromGame)
21:17:11 INFO     sc.networking.clients.LobbyClient - Received RemovedFromGame
21:17:11 INFO     sc.networking.clients.LobbyClient - Left 346cf859-52c6-48d3-911a-0c0096bebcf0
21:17:11 WARN   sc.networking.clients.XStreamClient - Writing on a closed Stream -> dropped the packet (tried to send package of type RemovedFromGame)
21:18:10 INFO   sc.server.network.NewClientListener - Shutting down NewClientListener...
21:18:10 DEBUG                        sc.gui.GuiApp - GuiApp stopped, ending program
```

--- a/gradle/build.gradle.kts
+++ b/gradle/build.gradle.kts
@@ -303,7 +303,17 @@ allprojects {
             test { testLogging { showStandardStreams = project.properties["verbose"] != null } }
             withType<Jar> {
                 if (plugins.hasPlugin(ApplicationPlugin::class))
-                    manifest.attributes["Main-Class"] = project.extensions.getByType<JavaApplication>().mainClass.get()
+                    manifest.attributes(
+                            "Main-Class" to project.extensions.getByType<JavaApplication>().mainClass.get(),
+                            "Add-Opens" to arrayOf(
+                                    "javafx.controls/javafx.scene.control.skin",
+                                    "javafx.controls/javafx.scene.control",
+                                    "javafx.graphics/javafx.scene",
+                                    // For accessing InputMap used in RangeSliderBehavior
+                                    "javafx.controls/com.sun.javafx.scene.control.inputmap",
+                                    // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
+                                    "java.base/java.util").joinToString(" ")
+                    )
             }
         }
     }

--- a/gradle/build.gradle.kts
+++ b/gradle/build.gradle.kts
@@ -112,18 +112,20 @@ tasks {
     }
     
     // TODO create a global constant which can be shared with testclient & co - maybe a resource?
-    val maxGameLength = 150L
+    val maxGameLength = 150L // 2m30s
     
     val testGame by creating {
-        dependsOn(":server:deploy", ":player:deployJar")
+        dependsOn(":server:deploy", ":player:deployShadow")
         group = "verification"
         doFirst {
             val testGameDir = testingDir.resolve("game")
             testGameDir.deleteRecursively()
             testGameDir.mkdirs()
+            val java = "java"
+                    //File("/usr/lib/jvm").listFiles { f:File -> f.name.contains("java-1") }?.max()?.resolve("bin/java").toString()
             val server =
                 ProcessBuilder(
-                    "java",
+                    java,
                     "-Dlogback.configurationFile=${project(":server").projectDir.resolve("configuration/logback-trace.xml")}",
                     "-jar", (project(":server").getTasksByName("jar", false).single() as Jar).archiveFile.get().asFile.absolutePath
                 )
@@ -134,7 +136,7 @@ tasks {
             Thread.sleep(400)
             val startClient: (Int) -> Process = {
                 Thread.sleep(100)
-                ProcessBuilder("java", "-jar", deployDir.resolve(deployedPlayer).absolutePath)
+                ProcessBuilder(java, "-jar", deployDir.resolve(deployedPlayer).absolutePath)
                     .redirectOutput(testGameDir.resolve("client$it.log")).redirectError(testGameDir.resolve("client$it-err.log")).start()
             }
             startClient(1)

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.9.1-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/helpers/test-client/build.gradle.kts
+++ b/helpers/test-client/build.gradle.kts
@@ -29,23 +29,12 @@ tasks {
     jar {
         dependsOn(createStartScripts)
         doFirst {
-            manifest {
-                attributes(
-                        "Class-Path" to
-                                configurations.default.get()
-                                        .map { "lib/" + it.name }
-                                        .plus("server.jar")
-                                        .joinToString(" "),
-                        "Add-Opens" to arrayOf(
-                                "javafx.controls/javafx.scene.control.skin",
-                                "javafx.controls/javafx.scene.control",
-                                "javafx.graphics/javafx.scene",
-                                // For accessing InputMap used in RangeSliderBehavior
-                                "javafx.controls/com.sun.javafx.scene.control.inputmap",
-                                // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
-                                "java.base/java.util").joinToString(" ")
-                )
-            }
+            manifest.attributes(
+                    "Class-Path" to configurations.default.get()
+                            .map { "lib/" + it.name }
+                            .plus("server.jar")
+                            .joinToString(" ")
+            )
             copy {
                 from("src/logback-tests.xml")
                 into(destinationDirectory)
@@ -61,7 +50,7 @@ tasks {
                 "--start-server --tests 3 --player1 $playerLocation --player2 $playerLocation"
             })
             @Suppress("UNNECESSARY_SAFE_CALL", "SimplifyBooleanWithConstants")
-            if (args?.isEmpty() == false)
+            if(args?.isEmpty() == false)
                 println("Using command-line arguments: $args")
         }
     }

--- a/helpers/test-client/build.gradle.kts
+++ b/helpers/test-client/build.gradle.kts
@@ -25,15 +25,27 @@ tasks {
         fileName = "start-tests"
         content = "java -Dfile.encoding=UTF-8 -Dlogback.configurationFile=logback-tests.xml -jar test-client.jar"
     }
-
+    
     jar {
         dependsOn(createStartScripts)
         doFirst {
-            manifest.attributes["Class-Path"] =
-                configurations.default.get()
-                        .map { "lib/" + it.name }
-                        .plus("server.jar")
-                        .joinToString(" ")
+            manifest {
+                attributes(
+                        "Class-Path" to
+                                configurations.default.get()
+                                        .map { "lib/" + it.name }
+                                        .plus("server.jar")
+                                        .joinToString(" "),
+                        "Add-Opens" to arrayOf(
+                                "javafx.controls/javafx.scene.control.skin",
+                                "javafx.controls/javafx.scene.control",
+                                "javafx.graphics/javafx.scene",
+                                // For accessing InputMap used in RangeSliderBehavior
+                                "javafx.controls/com.sun.javafx.scene.control.inputmap",
+                                // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
+                                "java.base/java.util").joinToString(" ")
+                )
+            }
             copy {
                 from("src/logback-tests.xml")
                 into(destinationDirectory)

--- a/player/build.gradle.kts
+++ b/player/build.gradle.kts
@@ -30,18 +30,6 @@ tasks {
     shadowJar {
         group = "distribution"
         archiveFileName.set("defaultplayer.jar")
-        manifest {
-            attributes(
-                    "Add-Opens" to arrayOf(
-                            "javafx.controls/javafx.scene.control.skin",
-                            "javafx.controls/javafx.scene.control",
-                            "javafx.graphics/javafx.scene",
-                            // For accessing InputMap used in RangeSliderBehavior
-                            "javafx.controls/com.sun.javafx.scene.control.inputmap",
-                            // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
-                            "java.base/java.util").joinToString(" ")
-            )
-        }
     }
     
     val copyDocs by creating(Copy::class) {
@@ -116,7 +104,7 @@ tasks {
         
         timeout.set(Duration.ofMinutes(1))
         workingDir(execDir)
-        executable = "./gradlew${if (OperatingSystem.current().isWindows) ".bat" else ""}"
+        executable = "./gradlew${if(OperatingSystem.current().isWindows) ".bat" else ""}"
         args("shadowJar", "--quiet", "--offline")
         
         doLast {

--- a/player/build.gradle.kts
+++ b/player/build.gradle.kts
@@ -18,7 +18,7 @@ sourceSets.main {
 }
 
 application {
-    mainClassName = "sc.player2021.Starter"
+    mainClassName = "sc.player2022.Starter"
 }
 
 dependencies {
@@ -30,6 +30,18 @@ tasks {
     shadowJar {
         group = "distribution"
         archiveFileName.set("defaultplayer.jar")
+        manifest {
+            attributes(
+                    "Add-Opens" to arrayOf(
+                            "javafx.controls/javafx.scene.control.skin",
+                            "javafx.controls/javafx.scene.control",
+                            "javafx.graphics/javafx.scene",
+                            // For accessing InputMap used in RangeSliderBehavior
+                            "javafx.controls/com.sun.javafx.scene.control.inputmap",
+                            // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
+                            "java.base/java.util").joinToString(" ")
+            )
+        }
     }
     
     val copyDocs by creating(Copy::class) {

--- a/player/src/main/sc/player2022/Starter.java
+++ b/player/src/main/sc/player2022/Starter.java
@@ -68,7 +68,7 @@ public class Starter {
     try {
       new Starter(host, port, reservation, room);
     } catch (Exception e) {
-      logger.error("Beim Starten den Clients ist ein Fehler aufgetreten:", e);
+      logger.error("Beim Starten des Clients ist ein Fehler aufgetreten:", e);
       e.printStackTrace();
       System.exit(1);
     }

--- a/player/src/main/sc/player2022/Starter.java
+++ b/player/src/main/sc/player2022/Starter.java
@@ -8,7 +8,6 @@ import sc.networking.clients.LobbyClient;
 import sc.player.IGameHandler;
 import sc.player.IPlayerClient;
 import sc.player2022.logic.Logic;
-import sc.plugin2022.util.Constants;
 import sc.shared.SharedConfiguration;
 
 import java.io.File;
@@ -24,7 +23,7 @@ public class Starter {
   public Starter(String host, int port, String reservation, String roomId) throws IOException {
     // Strategie zuweisen
     IGameHandler logic = new Logic();
-    IPlayerClient client = new LobbyClient(host, port).asPlayer(logic);
+    IPlayerClient client = new LobbyClient(host, port).asPlayer(logic, false);
 
     // einem Spiel beitreten
     if (reservation != null && !reservation.isEmpty()) {
@@ -71,8 +70,8 @@ public class Starter {
     } catch (Exception e) {
       logger.error("Beim Starten den Clients ist ein Fehler aufgetreten:", e);
       e.printStackTrace();
+      System.exit(1);
     }
-
   }
 
   private static void showHelp(String errorMsg) {

--- a/sdk/src/main/server-api/sc/networking/clients/AbstractLobbyClientListener.kt
+++ b/sdk/src/main/server-api/sc/networking/clients/AbstractLobbyClientListener.kt
@@ -13,7 +13,6 @@ abstract class AbstractLobbyClientListener: ILobbyClientListener {
     
     override fun onError(error: ErrorPacket) {}
     override fun onGamePrepared(response: GamePreparedResponse) {}
-    override fun onGameLeft(roomId: String) {}
     override fun onGameJoined(roomId: String) {}
     override fun onGameObserved(roomId: String) {}
 }

--- a/sdk/src/main/server-api/sc/networking/clients/ILobbyClientListener.kt
+++ b/sdk/src/main/server-api/sc/networking/clients/ILobbyClientListener.kt
@@ -15,7 +15,6 @@ interface ILobbyClientListener {
     fun onError(error: ErrorPacket)
     
     fun onGamePrepared(response: GamePreparedResponse)
-    fun onGameLeft(roomId: String)
     fun onGameJoined(roomId: String)
     fun onGameObserved(roomId: String)
 }

--- a/sdk/src/main/server-api/sc/networking/clients/LobbyClient.java
+++ b/sdk/src/main/server-api/sc/networking/clients/LobbyClient.java
@@ -37,6 +37,8 @@ public final class LobbyClient extends XStreamClient implements IPollsHistory {
   private final List<IHistoryListener> historyListeners = new ArrayList<>();
 
   private Function<RoomMessage, RoomMessage> player = null;
+  /** Whether to preserve the connection when a game ends. */
+  private boolean keepAlive = true;
   private Consumer<ResponsePacket> administrativeListener = null;
   private final Map<String, Consumer<ObservableRoomMessage>> roomObservers = new HashMap<>();
 
@@ -55,7 +57,6 @@ public final class LobbyClient extends XStreamClient implements IPollsHistory {
     return new AdminClient(this);
   }
 
-
   /** Sets observer to observe messages in the given room.
    * Whether administrative messages are received depends on authentication,
    * which has to be done separately. */
@@ -64,16 +65,24 @@ public final class LobbyClient extends XStreamClient implements IPollsHistory {
   }
 
   /** Sets this client up to play a game utilizing the handler.
+   * @param handler the game logic
    * @return a PlayerClient to join a room as Player. */
-  public @NotNull IPlayerClient asPlayer(IGameHandler handler) {
+  public @NotNull IPlayerClient asPlayer(IGameHandler handler) {return asPlayer(handler, false);}
+
+  /** Sets this client up to play games utilizing the handler.
+   * @param handler the game logic
+   * @param keepAlive whether the client should stay connected after a game ends
+   * @return a PlayerClient to join a room as Player. */
+  public @NotNull IPlayerClient asPlayer(IGameHandler handler, boolean keepAlive) {
     start();
     PlayerClient client = new PlayerClient(this, handler);
     player = client;
+    this.keepAlive = keepAlive;
     return client;
   }
 
   @Override
-  protected final void onObject(ProtocolPacket message) {
+  protected void onObject(ProtocolPacket message) {
     if(message instanceof ResponsePacket && administrativeListener != null)
       administrativeListener.accept((ResponsePacket) message);
     if (message instanceof RoomPacket) {
@@ -102,14 +111,17 @@ public final class LobbyClient extends XStreamClient implements IPollsHistory {
       } else {
         onRoomMessage(roomId, data);
       }
+    } else if (message instanceof RemovedFromGame) {
+      String roomId = ((RemovedFromGame) message).getRoomId();
+      logger.info("Left {}", roomId);
+      if(!keepAlive)
+        stop();
     } else if (message instanceof GamePreparedResponse) {
       onGamePrepared((GamePreparedResponse) message);
     } else if (message instanceof JoinedRoomResponse) {
       onGameJoined(((JoinedRoomResponse) message).getRoomId());
     } else if (message instanceof RoomWasJoinedEvent) {
       onGameJoined(((RoomWasJoinedEvent) message).getRoomId());
-    } else if (message instanceof RemovedFromGame) {
-      onGameLeft(((RemovedFromGame) message).getRoomId());
     } else if (message instanceof ObservationResponse) {
       onGameObserved(((ObservationResponse) message).getRoomId());
     } else if (message instanceof ErrorPacket) {
@@ -138,14 +150,6 @@ public final class LobbyClient extends XStreamClient implements IPollsHistory {
     for (ILobbyClientListener listener : this.listeners) {
       listener.onGameOver(roomId, data);
     }
-  }
-
-  private void onGameLeft(String roomId) {
-    logger.info("Received RemovedFromGame");
-    for (ILobbyClientListener listener : this.listeners) {
-      listener.onGameLeft(roomId);
-    }
-    logger.info("Left {}", roomId);
   }
 
   private void onGameJoined(String roomId) {

--- a/sdk/src/main/server-api/sc/networking/clients/XStreamClient.java
+++ b/sdk/src/main/server-api/sc/networking/clients/XStreamClient.java
@@ -44,7 +44,7 @@ public abstract class XStreamClient implements IClient {
     /** disconnected because CloseConnection was received (disconnected by other side) */
     RECEIVED_DISCONNECT,
     /** disconnected from this side */
-    DISCONNECTED,
+    INITIATED_DISCONNECTED,
     // error conditions
     PROTOCOL_ERROR,
     LOST_CONNECTION,
@@ -234,7 +234,7 @@ public abstract class XStreamClient implements IClient {
     // this side caused disconnect, notify other side
     if (!isClosed())
       send(new CloseConnection());
-    handleDisconnect(DisconnectCause.DISCONNECTED);
+    handleDisconnect(DisconnectCause.INITIATED_DISCONNECTED);
   }
 
   protected synchronized void stopReceiver() {

--- a/sdk/src/main/server-api/sc/networking/clients/XStreamClient.java
+++ b/sdk/src/main/server-api/sc/networking/clients/XStreamClient.java
@@ -44,7 +44,7 @@ public abstract class XStreamClient implements IClient {
     /** disconnected because CloseConnection was received (disconnected by other side) */
     RECEIVED_DISCONNECT,
     /** disconnected from this side */
-    INITIATED_DISCONNECTED,
+    INITIATED_DISCONNECT,
     // error conditions
     PROTOCOL_ERROR,
     LOST_CONNECTION,
@@ -234,7 +234,7 @@ public abstract class XStreamClient implements IClient {
     // this side caused disconnect, notify other side
     if (!isClosed())
       send(new CloseConnection());
-    handleDisconnect(DisconnectCause.INITIATED_DISCONNECTED);
+    handleDisconnect(DisconnectCause.INITIATED_DISCONNECT);
   }
 
   protected synchronized void stopReceiver() {

--- a/server/build.gradle.kts
+++ b/server/build.gradle.kts
@@ -7,7 +7,8 @@ plugins {
 application {
     mainClass.set("sc.server.Application")
     applicationDefaultJvmArgs = listOf("-Dfile.encoding=UTF-8",
-            "-XX:+PrintGC", "-XX:+PrintGCDetails", "-XX:+PrintGCDateStamps", "-Xloggc:gc.log")
+            "-XX:+PrintGC", "-XX:+PrintGCDetails", "-XX:+PrintGCDateStamps", "-Xloggc:gc.log" // GC
+    )
 }
 
 dependencies {
@@ -92,7 +93,19 @@ tasks {
     jar {
         destinationDirectory.set(runnableDir)
         doFirst {
-            manifest.attributes["Class-Path"] = configurations.default.get().joinToString(" ") { "lib/" + it.name }
+            manifest {
+                attributes(
+                        "Class-Path" to configurations.default.get().joinToString(" ") { "lib/" + it.name },
+                        "Add-Opens" to
+                                arrayOf("javafx.controls/javafx.scene.control.skin",
+                                        "javafx.controls/javafx.scene.control",
+                                        "javafx.graphics/javafx.scene",
+                                        // For accessing InputMap used in RangeSliderBehavior
+                                        "javafx.controls/com.sun.javafx.scene.control.inputmap",
+                                        // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
+                                        "java.base/java.util").joinToString(" ")
+                )
+            }
         }
     }
     

--- a/server/build.gradle.kts
+++ b/server/build.gradle.kts
@@ -76,10 +76,11 @@ tasks {
         group = "application"
         dependsOn(makeRunnable)
         workingDir = buildDir
-		doFirst {
-			val tag = Runtime.getRuntime().exec(arrayOf("git", "rev-parse", "--short", "--verify", "HEAD")).inputStream.reader().readText().trim()
-			val relativeRunnable = runnableDir.relativeTo(workingDir)
-			commandLine("docker", "build", "--no-cache", "-t", "swc_game-server:latest", "-t", "swc_game-server:$tag", "--build-arg", "game_server_dir=$relativeRunnable", ".")
+        doFirst {
+            val tag = Runtime.getRuntime().exec(arrayOf("git", "rev-parse", "--short", "--verify", "HEAD"))
+                    .inputStream.reader().readText().trim()
+            val relativeRunnable = runnableDir.relativeTo(workingDir)
+            commandLine("docker", "build", "--no-cache", "-t", "swc_game-server:latest", "-t", "swc_game-server:$tag", "--build-arg", "game_server_dir=$relativeRunnable", ".")
             copy {
                 from(projectDir.resolve("configuration"))
                 include("?ocker*")
@@ -87,25 +88,15 @@ tasks {
                 include("server.properties.production")
                 into(workingDir)
             }
-		}
+        }
     }
     
     jar {
         destinationDirectory.set(runnableDir)
         doFirst {
-            manifest {
-                attributes(
-                        "Class-Path" to configurations.default.get().joinToString(" ") { "lib/" + it.name },
-                        "Add-Opens" to
-                                arrayOf("javafx.controls/javafx.scene.control.skin",
-                                        "javafx.controls/javafx.scene.control",
-                                        "javafx.graphics/javafx.scene",
-                                        // For accessing InputMap used in RangeSliderBehavior
-                                        "javafx.controls/com.sun.javafx.scene.control.inputmap",
-                                        // Expose list internals for xstream conversion: https://github.com/x-stream/xstream/issues/253
-                                        "java.base/java.util").joinToString(" ")
-                )
-            }
+            manifest.attributes(
+                    "Class-Path" to configurations.default.get().joinToString(" ") { "lib/" + it.name }
+            )
         }
     }
     

--- a/server/src/test/java/sc/server/client/TestLobbyClientListener.java
+++ b/server/src/test/java/sc/server/client/TestLobbyClientListener.java
@@ -14,7 +14,6 @@ public class TestLobbyClientListener implements ILobbyClientListener {
   public boolean gamePausedReceived = false;
   public boolean gameOverReceived = false;
   public boolean gameJoinedReceived = false;
-  public boolean gameLeftReceived = false;
   public boolean gamePreparedReceived = false;
   public boolean roomMessageReceived = false;
   public boolean errorReceived = false;
@@ -37,7 +36,6 @@ public class TestLobbyClientListener implements ILobbyClientListener {
             gamePausedReceived == o.gamePausedReceived &&
             gameOverReceived == o.gameOverReceived &&
             gameJoinedReceived == o.gameJoinedReceived &&
-            gameLeftReceived == o.gameLeftReceived &&
             gamePreparedReceived == o.gamePreparedReceived &&
             roomMessageReceived == o.roomMessageReceived &&
             errorReceived == o.errorReceived &&
@@ -74,12 +72,6 @@ public class TestLobbyClientListener implements ILobbyClientListener {
   }
 
   @Override
-  public void onGameLeft(String roomId) {
-    this.roomId = roomId;
-    gameLeftReceived = true;
-  }
-
-  @Override
   public void onGameJoined(String roomId) {
     this.roomId = roomId;
     gameJoinedReceived = true;
@@ -105,7 +97,6 @@ public class TestLobbyClientListener implements ILobbyClientListener {
             ", gamePausedReceived=" + gamePausedReceived +
             ", gameOverReceived=" + gameOverReceived +
             ", gameJoinedReceived=" + gameJoinedReceived +
-            ", gameLeftReceived=" + gameLeftReceived +
             ", gamePreparedReceived=" + gamePreparedReceived +
             ", roomMessageReceived=" + roomMessageReceived +
             ", errorReceived=" + errorReceived +

